src/dash-table/syntax-tree/lexeme/relational.ts

```diff
@@ -50,12 +50,16 @@ const LEXEME_BASE = {
 };
 
 export const contains: IUnboundedLexeme = R.merge({
-    evaluate: relationalEvaluator(([op, exp]) =>
-        !R.isNil(exp) &&
-        !R.isNil(op) &&
-        (R.type(exp) === 'String' || R.type(op) === 'String') &&
-        op.toString().indexOf(exp.toString()) !== -1
-    ),
+    evaluate: relationalEvaluator(([op, exp]) => {
+        if (R.isNil(op) || R.isNil(exp)) {
+            return false;
+        }
+
+        op = typeof op === 'number' ? op.toString() : op;
+        exp = typeof exp === 'number' ? exp.toString() : exp;
+
+        return op.toString().indexOf(exp.toString()) !== -1;
```

Comment:
isn't this double `.toString`?

But also: should we be trying to use the column formatting here? What if you have prices formatted with 2-digit decimals and you want to match all even-dollar values, with `contains '.00'`?

Reply:
Getting access to formatting from here would be difficult. In a sense this would be no different from when we do a number comparison where the number could be presented in hex or the expression be an hex.

I think you're right, I got carried away by the number case for dates. Deviating too much from what was agreed upon when we last went through filtering. Will revert the `contains` changes and focus on (1) datestartswith and (2) the lexeme bug

Reply:
OK - certainly understand that this would be tough, and it's a weird edge case so I'm happy to ignore for now. Let's make an issue for it though, if it doesn't already exist. If a user is filtering as a string, they would expect the string used in the filter to be the one they see.